serde-reflection/src/format.rs

```diff
@@ -261,6 +321,92 @@ where
         }
         self.value.unify(other.value)
     }
+
+    fn is_unknown(&self) -> bool {
+        false
+    }
+}
+
+impl<T> Variable<T> {
+    pub(crate) fn new(content: Option<T>) -> Self {
+        Self(Rc::new(RefCell::new(content)))
+    }
+
+    fn borrow(&self) -> Ref<Option<T>> {
+        self.0.as_ref().borrow()
+    }
+
+    fn borrow_mut(&mut self) -> RefMut<Option<T>> {
+        self.0.as_ref().borrow_mut()
+    }
```

Comment:
nit: maybe use the Borrow/BorrowMut traits?
(i'm fine if you do leave this as is)

Reply:
I tried and unfortunately (or fortunately for type safety of Rust :)), the following code doesn't work: 
```
fn borrow(&self) -> &Option<T> {  // <- & instead of Ref is required by the trait
    self.0.as_ref().borrow().deref() // but now we cannot delay the "unborrowing" of the refcell so lifetime checks fail.
}
```